tests/manage/pv_services/pvc_snapshot/test_pvc_snapshot_when_snapshotclass_deleted.py

```diff
@@ -0,0 +1,148 @@
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier3,
+    skipif_ocs_version,
+    skipif_ocp_version,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources import pvc, ocs
+from ocs_ci.helpers import helpers
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+
+@tier3
+@skipif_ocs_version("<4.7")
+@skipif_ocp_version("<4.7")
+@pytest.mark.polarion_id("OCS-2638")
+class TestPvcSnapshotWhenSnapshotClassDeleted(ManageTest):
+    """
+    Tests to verify PVC snapshot feature when snapshotclass deleted
+    """
+
+    @pytest.fixture(autouse=True)
+    def base_setup(self, interface_iterate, pvc_factory, pod_factory):
+        """
+        Create resources for the test
+
+        Args:
+            interface_iterate: A fixture to iterate over interfaces
+            pvc_factory: A fixture to create new pvc
+            pod_factory: A fixture to create new pod
+
+        """
+
+        # Create resources for the test
+        self.interface = interface_iterate
+
+        self.pvc_obj = pvc_factory(
+            interface=self.interface, size=5, status=constants.STATUS_BOUND
+        )
+
+        self.pod_object = pod_factory(
+            interface=self.interface, pvc=self.pvc_obj, status=constants.STATUS_RUNNING
+        )
+
+    def create_snapshotclass(self, interface):
+        """
+        Creates own volumesnapshotclass
+
+        Args:
+            interface (str): Interface type used
+
+        """
+        if interface == constants.CEPHFILESYSTEM:
+            snapshotclass_data = templating.load_yaml(
+                constants.CSI_CEPHFS_SNAPSHOTCLASS_YAML
+            )
+            snapclass_name = "cephfssnapshotclass"
+        else:
+            snapshotclass_data = templating.load_yaml(
+                constants.CSI_RBD_SNAPSHOTCLASS_YAML
+            )
+            snapclass_name = "rbdsnapshotclass"
+        snapshotclass_data["metadata"]["name"] = snapclass_name
+        ocs_obj = ocs.OCS(**snapshotclass_data)
+        created_snapclass = ocs_obj.create(do_reload=True)
+        assert created_snapclass, f"Failed to create snapshot {snapclass_name}"
+        return ocs_obj
+
+    def test_pvc_snapshot(self, interface_iterate, teardown_factory):
+        """
+        1. Create PVC, POD's
+        2. Create own volumesnapshotclass
+        3. Take a snapshot of PVC from created snapshotclass
+        4. Delete the volumesnapshotclass used to create the above volume snapshot
+        5. Check the status of volume snapshot
+
+        """
+
+        # Create own snapshotclass
+        snapclass_obj = self.create_snapshotclass(interface=self.interface)
+        teardown_factory(snapclass_obj)
+
+        # Take a snapshot from created snapshotclass
+        snap_yaml = constants.CSI_RBD_SNAPSHOT_YAML
+        if self.interface == constants.CEPHFILESYSTEM:
+            snap_yaml = constants.CSI_CEPHFS_SNAPSHOT_YAML
```

Comment:
This can be replaced with,
snap_yaml = constants.CSI_CEPHFS_SNAPSHOT_YAML if self.interface == constants.CEPHFILESYSTEM else constants.CSI_RBD_SNAPSHOT_YAML